lib/srv/server/azure_watcher.go

```diff
@@ -97,7 +97,7 @@ func NewAzureWatcher(ctx context.Context, fetchersFn func() []Fetcher, opts ...O
 }
 
 // MatchersToAzureInstanceFetchers converts a list of Azure VM Matchers into a list of Azure VM Fetchers.
-func MatchersToAzureInstanceFetchers(matchers []types.AzureMatcher, clients azureClientGetter) []Fetcher {
+func MatchersToAzureInstanceFetchers(matchers []types.AzureMatcher, clients azureClientGetter, discoveryConfig string) []Fetcher {
```

Comment:
Tbh I'm not a huge fan of the fact that we need to propagate this discovery config as an extra parameter everywhere. Seems very tightly coupled. Is there a cleaner way to achieve this?

Reply:
DiscoveryConfig contains Matchers, which create Fetchers.
Fetchers find resources and enrolls them. When it fails, we must update the DiscoveryConfigStatus / UserTasks and we need the DiscoveryConfig name.

So, what I'm doing is adding the DiscoveryConfig name into the Fetcher, so that when it tries to auto-enroll resources and it fails, it is able to identify which DiscoveryConfig was used.

I don't think there's a way to achieve this without passing the DiscoveryConfig name into the Fetcher.
We can create a new type that includes the DiscoveryConfigName as well as the Matcher. Something like this:
```go
type MatcherFromDiscoveryConfig struct {
    DiscoveryConfigName string
    AzureMatcher types.AzureMatcher
    AWSMatcher types.AWSMatcher
    GCPMatcher types.GCPMatcher
}
```
And when creating the Fetcher, we would get the DiscoveryConfigName from this new type, instead of as a param.
I'm not too fond of this, and would rather keep it as is.
Please let me know what you think.